core/modules/simpletest/tests/common.test

```diff
@@ -875,6 +875,41 @@ class CommonCascadingStylesheetsTestCase extends BackdropWebTestCase {
     $this->assert(strpos($styles, $simpletest . '/tests/system.css') === FALSE, 'The overridden CSS file is not output.');
   }
 
+  /**
+   * Test CSS Styles with attributes rendering.
+   *
+   * @todo make it more like JavaScriptTestCase::testAttributes() {
+   */
+  function testRenderStylesAttributes() {
+    // Disable aggregation.
+    config_set('system.core', 'preprocess_css', 0);
+
+    backdrop_add_css(
+      'misc/print.css',
+      array(
+        'attributes' => array(
+          'foo' => 'bar',
+        ),
+      )
+    );
+
+    // Retrieve the rendered Styles.
+    $css = backdrop_get_css();
+
+    $expected = array(
+      'foo' => 'bar',
+      'media' => 'all',
+    );
+
+    foreach ($expected as $expected_string) {
+      $this->assertNotEqual(
+        strpos($css, $expected_string),
+        FALSE,
+        'Styles with attributes has been properly rendered.'
+      );
+    }
```

Comment:
I wanted to see how this differed from `CommonJavaScriptTestCase::testAttributes()` since it's mentioned in the docblock. Here's that code:

```
    // Only 'custom' attribute is set.
    backdrop_add_js($js_internal, array('attributes' => array('custom' => 'foo')));
    backdrop_add_js($js_external, array('attributes' => array('custom' => 'foo'), 'type' => 'external'));
    $javascript = backdrop_get_js();
    $this->backdropSetContent($javascript);
    $this->assertTrue($this->xpath('//script[starts-with(@src, "' . file_create_url($js_internal) . '") and @custom="foo"]'), 'Rendered internal JavaScript with correct custom attribute.');
    $this->assertTrue($this->xpath('//script[@src="' . $js_external . '" and @custom="foo"]'), 'Rendered external JavaScript with correct custom attribute.');
```

The JS testing approach does seem to be better and not too much effort. The current test seems a little shy of effective testing because it's only testing if the value is present, not the attribute name.

Reply:
Easy peasy. I've updated the PR.